### DISCUSSION TITLE
14.0.X and Master branch: TransactionConfigurationBuilder is not compatible with Java 17

### DISCUSSION THREAD

**Discussion by jmmain69** [infinispan/infinispan#10712] [General]
`core/src/main/java/org/infinispan/configuration/cache/TransactionConfigurationBuilder.java` is importing `javax.transaction` classes.
```
import javax.transaction.Synchronization;
import javax.transaction.TransactionManager;
import javax.transaction.xa.XAResource;
```

The `core/pom.xml` defines a `jakarta.transaction` dependency.  
```
      <dependency>
         <groupId>jakarta.transaction</groupId>
         <artifactId>jakarta.transaction-api</artifactId>
      </dependency>
```

All `javax.transaction` class imports need to be changed to `jakarta.transaction` for Java 17 compatibility.

**Comment by pruivo** [comment]
That is intentional. We have 2 "core" jars available:
* `infinispan-core` imports `javax` package
* `infinispan-core-jarkarta` imports `jakarta` package: [Maven Central](https://repo1.maven.org/maven2/org/infinispan/infinispan-core-jakarta/14.0.7.Final/)

**Reply by jmmain69** [reply]
What would be the correct way the import that when using 
```
        <dependency>
            <groupId>org.infinispan</groupId>
            <artifactId>infinispan-spring-boot-starter-embedded</artifactId>
        </dependency>
 ```

**Reply by jmmain69** [reply]
I appreciate the swift response.  Looks like `infinispan-spring-boot3-starter-embedded` is what we need now.
